app/actions/wrap.ts

```diff
@@ -12,7 +13,7 @@ export const changeApprovalTransaction = async (params: {
       contractName: "sklima",
       provider: params.provider.getSigner(),
     });
-    const value = ethers.utils.parseUnits("1000000000", "gwei"); //bignumber
+    const value = ethers.utils.parseUnits(params.value, 9); // bignumber, is 9 correct here?
```

Comment:
I had to think this through, a bit confusing:
we only care about decimals of the approving contract, i.e. the token that will be spent. in this case thats sklima.approve() so, 9 is correct

Reply:
Thanks, comment is removed